src/Windows/Host/Client/Impl/BrokerServices/LocalStaticFileServer.cs

```diff
@@ -26,19 +26,17 @@ internal class LocalStaticFileServer : StaticFileServerBase {
         }
 
         public override async Task HandleRequestAsync(HttpListenerContext context, CancellationToken ct) {
-            UriBuilder ub = new UriBuilder();
+            UriBuilder ub = new UriBuilder(context.Request.Url);
```

Comment:
This will likely copy more than you want - e.g. port and username/password. Safer to build it only from the components that you want there, like you were doing before. I think the only thing that you do need from the original URL is `Path`, and then set scheme and host to fixed values.